### ISSUE TITLE
Add role checking for tasks

### ISSUE THREAD

**Issue by dr-dobermann** [dr-dobermann/gobpm#29]
Task uses Role resources to control access. 
So there is necessity to create a framework to support access check to tasks.

**Comment by dr-dobermann** [comment]
Access to resources should be controlled by ACL.

**Comment by dr-dobermann** [comment]
Superseded by the landed UserTask authorization (ADR-020 / SRD-034, merged 2026-07-03): task access is gated by the Camunda-triad assignments — assignee / candidateUsers / candidateGroups — via
typed `Assignments()` and `Actor{UserID, Groups}`, enforced on both `Take` and `Complete`. The 2024 'Role resources' concept this issue described was explicitly replaced by that model, so there is nothing left to
implement under this framing.

**Comment by dr-dobermann** [comment]
Closed as **superseded / completed by another route**.

The 2024 "Role resources" concept this issue was written against was replaced
during the User Task work: **ADR-020 / SRD-034** landed Camunda-triad
authorization for tasks — `assignee` / `candidateUsers` / `candidateGroups`
(typed `hinteraction.Assignment`, reached via `Assignments()`), gating both
`Take` and `Complete` with a typed `Actor{UserID, Groups}`. The engine-side
access-check framework this issue asked for exists; the `Roles()`-projection
idea was explicitly considered and rejected in that design.

Anything beyond the triad (e.g. policy engines, IAM integration) is tracked by
the **Multi-tenancy and IAM** epic (#73), not this task.
